lite-pm/pages/components/AppPage.js

```diff
@@ -228,23 +218,41 @@ class AppPage extends Component {
     }
 
     const projectId = this.state.project.projectId;
-    axios.post(`${baseUrl}/api/tasks/create`, {
-      projectId: projectId,
-      title: this.state.taskName,
-      duration: this.state.taskDuration,
-      description: this.state.taskDescription,
-    });
-
-    this.setState(
-      {
-        Task: tempTasks,
-        taskName: "Name",
-        taskDescription: "Description",
-        taskDuration: 0,
-        taskDurationType: 0,
-      },
-      () => this.refreshTasks()
-    );
+    axios
+      .post(`${baseUrl}/api/tasks/create`, {
+        projectId: projectId,
+        title: this.state.taskName,
+        duration: this.state.taskDuration,
+        description: this.state.taskDescription,
+      })
+      .then((id) => {
+        tempTasks.push({
+          handleStop: this.handleStop,
+          key: uuid(),
+          taskId: id.data,
+          index: tempTasks.length,
+          title: this.state.taskName,
+          status: "todo",
+          description: this.state.taskDescription,
+          duration: this.state.taskDuration,
+          durationType: this.state.taskDurationType,
+          userId: "",
+        });
+
+        this.setState(
+          {
+            Task: tempTasks,
+            taskName: "Name",
+            taskDescription: "Description",
+            taskDuration: 0,
+            taskDurationType: 0,
+          },
+          () => {
+            this.refreshTasks();
+            this.getProjectDetails(this.refreshTasks);
+          }
```

Comment:
If addTasks returns the current state of the project in the API (which I believe it does), then we can update the state based on that response instead of making another API call

Reply:
I'm unsure about this one, the api call after the refresh was to keep the database in sync with the frontend because otherwise weird behaviors occur usually